raiden/network/proxies/token_network.py

```diff
@@ -708,51 +657,89 @@ def set_total_deposit(
         """
         typecheck(total_deposit, int)
 
-        with self.channel_operations_lock[partner], self.deposit_lock:
-            previous_total_deposit = self._detail_participant(
-                channel_identifier=channel_identifier,
-                detail_for=self.node_address,
-                partner=partner,
-                block_identifier=given_block_identifier,
-            ).deposit
+        token_address = self.token_address()
+        token = Token(
+            jsonrpc_client=self.client,
+            token_address=token_address,
+            contract_manager=self.contract_manager,
+        )
 
-            token_address = self.token_address()
-            token = Token(
-                jsonrpc_client=self.client,
-                token_address=token_address,
-                contract_manager=self.contract_manager,
-            )
+        with self.channel_operations_lock[partner], self.deposit_lock:
             try:
-                self._deposit_preconditions(
+                channel_onchain_detail = self._detail_channel(
+                    participant1=self.node_address,
+                    participant2=partner,
+                    block_identifier=given_block_identifier,
+                    channel_identifier=channel_identifier,
+                )
+                sender_details = self._detail_participant(
                     channel_identifier=channel_identifier,
-                    total_deposit=total_deposit,
+                    detail_for=self.node_address,
                     partner=partner,
-                    token=token,
-                    previous_total_deposit=previous_total_deposit,
                     block_identifier=given_block_identifier,
                 )
-            except NoStateForBlockIdentifier:
-                # If preconditions end up being on pruned state skip them. Estimate
-                # gas will stop us from sending a transaction that will fail
+            except ValueError:
+                # If `given_block_identifier` has been pruned the checks cannot be
+                # performed.
                 pass
+            except BadFunctionCallOutput:
+                raise_on_call_returned_empty(given_block_identifier)
+            else:
```

Comment:
This is missing a checks for:
- [the participants maximum deposit amount](https://github.com/raiden-network/raiden-contracts/blob/8ff92c6ce4d23537b703d8f98f5ef2e8678b3add/raiden_contracts/data/source/raiden/TokenNetwork.sol#L324)
- [the network maximum deposit](https://github.com/raiden-network/raiden-contracts/blob/8ff92c6ce4d23537b703d8f98f5ef2e8678b3add/raiden_contracts/data/source/raiden/TokenNetwork.sol#L348)
- Would this [deposit cause an overflow to happen](https://github.com/raiden-network/raiden-contracts/blob/8ff92c6ce4d23537b703d8f98f5ef2e8678b3add/raiden_contracts/data/source/raiden/TokenNetwork.sol#L356)? Highly unlikely, but should be checked anyways (useful with testnet deployments when huge values are used)

Reply:
Done

Reply:
This is also missing the check for the deprecation switch: https://github.com/raiden-network/raiden-contracts/blob/acd697fec11be7e008abaf3abf638b1ceff8d253/raiden_contracts/data/source/raiden/TokenNetwork.sol#L182

Reply:
Done... this was done in pre-conditions check. Should it also be done in the post-conditions?